src/lurk/cli/meta.rs

```diff
@@ -1072,6 +1099,254 @@ impl<H: Chipset<F>> MetaCmd<F, H> {
+                            let next_callable_zptr = next_callable.commit(&mut repl.zstore);
+
+                            // the result is a pair composed by the chain result and next callable
+                            // provided by the client
+                            let result = repl
+                                .zstore
+                                .intern_cons(chain_result.zptr, next_callable_zptr);
+
+                            // and now the proof must verify, meaning that the user must have
+                            // used the correct callable from the server state
+                            let machine_proof =
+                                crypto_proof.into_machine_proof(&expr, &empty_env, &result);
+                            let machine = repl.stark_machine();
+                            let (_, vk) = machine.setup(&LairMachineProgram);
+                            let challenger = &mut machine.config().challenger();
+                            if machine.verify(&vk, &machine_proof, challenger).is_err() {
+                                write_data(
+                                    &mut stream,
+                                    Response::ProofVerificationFailed(
+                                        get_verifier_version().to_string(),
+                                    ),
+                                )?;
+                                continue;
+                            }
+
+                            // everything went okay... transition to the next state
+                            write_data(&mut stream, Response::ProofAccepted)?;
+                            state_chain_result = chain_result.populate_zstore(&mut repl.zstore);
+                            next_callable.populate_zstore(&mut repl.zstore);
+                            state_callable = next_callable_zptr;
+                        }
+                    },
+                    Err(e) => {
+                        eprintln!("Connection failed: {e}");
+                    }
+                }
+            }
+            Ok(())
+        },
+    };
+
+    const MICRO_CHAIN_GET: Self = Self {
+        name: "micro-chain-get",
+        summary: "Binds the current state from a micro-chain server to a symbol",
+        format: "!(micro-chain-get <addr_str> <symbol>)",
+        info: &[],
+        example: &["!(micro-chain-get \"127.0.0.1:1234\" state0)"],
+        run: |repl, args, _path| {
+            let (addr, &state_sym) = repl.peek2(args)?;
+            if addr.tag != Tag::Str {
+                bail!("Address must be a string");
+            }
+            Self::validate_binding_var(repl, &state_sym)?;
+            let addr_str = repl.zstore.fetch_string(addr);
+            let stream = &mut TcpStream::connect(addr_str)?;
+            write_data(stream, Request::Get)?;
+            let Response::State(chain_result, next_callable) = read_data(stream)? else {
+                unreachable!()
```

Comment:
nit
```suggestion
                bail!("Could not read state from server");
```
I think this could happen if you connect to something that isn't the micro-chain TCP server. If I'm wrong keeping `unreachable!` is fine